extensions/amp-video/0.1/test/test-video-cache.js

```diff
@@ -494,6 +502,61 @@ describes.realWin('amp-video cached-sources', {amp: true}, (env) => {
     });
   });
 
+  describe('captions_src field', async () => {
+    it('should append track element if the cache responds with captions_src', async () => {
+      env.sandbox.stub(xhrService, 'fetch').resolves({
+        json: () =>
+          Promise.resolve({
+            'captions_src': 'captions_src_response.vtt',
+            'captions_srclang': 'en-us',
+            'sources': [
+              {'url': 'video.mp4', 'bitrate_kbps': 700, 'type': 'video/mp4'},
+            ],
+          }),
+      });
+      const videoEl = createVideo([{src: 'video.mp4'}]);
+      await fetchCachedSources(videoEl, env.ampdoc);
+
+      const trackEl = videoEl.querySelector('track');
+      expect(trackEl).to.exist;
+    });
+    it('should not append track element if video already has a track child', async () => {
```

Comment:
(thank you for writing such great tests it makes reviewing the UX a lot easier)

maybe we should still render the track element after the existing ones? So if they 404 it falls back to ours

Reply:
to recap our conversation:
Track elements don't automatically fallback to the next track upon 404. 
For now we will keep this functionality of replacing the track.